### ISSUE TITLE
ResilioSync.munki.recipe downloads but doesn't add to Munki repo

### ISSUE THREAD

**Issue by fseesink** [autopkg/homebysix-recipes#87]
I've been using the BTSync recipe for some time, and finally got around to replacing it with the Resilio recipe (as in removing the BTSync Munki recipe and now using the Resilio Munki recipe).

Now I honestly don't know whether this is due to the Resilio recipe, **AutoPgr** (which I recently upgraded to 1.4.2), or my own stupidity (quite possible).  I haven't been modifying Munki configs in awhile so I'm a tad rusty.  But while all my other recipes seem to be working fine (and are updating apps in the repo/etc.), and I can see that this one is downloading the .DMG in the cache folder, it's not adding it to the repo.  And I can't for the life of me figure out why.  I've been doing some stare and compares of different recipes, and at least the .munki.recipe files look to be fine.  (One question:  there's the use of **%NAME%** in several places including pathnames, and some apps like **Resilio Sync** have a space in them.  Could this be the issue?  I've tried replacing **%NAME%** with **ResilioSync** (no space) but no dice.)

I sure wish AutoPkg/AutoPkgr would have a debug mode where we could see each step being executed to find out why things aren't going as they should.

Anyway, am I nuts here?  When I still had the BTSync munki recipe, I noticed at one point it started pulling down Resilio Sync and adding that to the repo.  I thought, "Oh, cool."  But it wasn't showing in the Managed Software Center where I had things configured, so figured it was time to clean up and just remove all references to BTSync and have it just be Resilio Sync.  (Mind you, my current setup only impacts machines I manage.)  But I'm clearly missing something.  At the very least, I can't seem to find logs/evidence to explain what I'm doing wrong.  But I'm hoping you might be able to point me in the right direction.

Thanks in advance for any/all help.

**Comment by homebysix** [comment]
Hi @fseesink,

I'll start with the less-than-helpful "works for me!"

```
$ autopkg run -v ResilioSync.munki.recipe
Processing ResilioSync.munki.recipe...
URLDownloader
URLDownloader: Storing new Last-Modified header: Thu, 29 Sep 2016 09:59:41 GMT
URLDownloader: Storing new ETag header: "ec7290f1933f836f747a020f200f53ee-3"
URLDownloader: Downloaded /Users/elliot/Library/AutoPkg/Cache/com.github.homebysix.munki.ResilioSync/downloads/Resilio Sync.dmg
EndOfCheckPhase
CodeSignatureVerifier
CodeSignatureVerifier: Mounted disk image /Users/elliot/Library/AutoPkg/Cache/com.github.homebysix.munki.ResilioSync/downloads/Resilio Sync.dmg
CodeSignatureVerifier: Verifying application bundle signature...
CodeSignatureVerifier: /private/tmp/dmg.7Q6v16/Resilio Sync.app: valid on disk
CodeSignatureVerifier: /private/tmp/dmg.7Q6v16/Resilio Sync.app: satisfies its Designated Requirement
CodeSignatureVerifier: /private/tmp/dmg.7Q6v16/Resilio Sync.app: explicit requirement satisfied
CodeSignatureVerifier: Signature is valid
Versioner
Versioner: Mounted disk image /Users/elliot/Library/AutoPkg/Cache/com.github.homebysix.munki.ResilioSync/downloads/Resilio Sync.dmg
Versioner: Found version 2.4.1 in file /private/tmp/dmg.X0Fwwb/Resilio Sync.app/Contents/Info.plist
MunkiImporter
MunkiImporter: Copied pkginfo to /Users/Shared/munki_repo/pkgsinfo/apps/Resilio Sync/Resilio Sync-2.4.1.plist
MunkiImporter: Copied pkg to /Users/Shared/munki_repo/pkgs/apps/Resilio Sync/Resilio Sync-2.4.1.dmg
Receipt written to /Users/elliot/Library/AutoPkg/Cache/com.github.homebysix.munki.ResilioSync/receipts/ResilioSync.munki-receipt-20161006-155354.plist

The following new items were imported into Munki:
    Name          Version  Catalogs  Pkginfo Path                                Pkg Repo Path                             
    ----          -------  --------  ------------                                -------------                             
    Resilio Sync  2.4.1    testing   apps/Resilio Sync/Resilio Sync-2.4.1.plist  apps/Resilio Sync/Resilio Sync-2.4.1.dmg  

The following new items were downloaded:
    Download Path                                                                                          
    -------------                                                                                          
    /Users/elliot/Library/AutoPkg/Cache/com.github.homebysix.munki.ResilioSync/downloads/Resilio Sync.dmg
```

The first thing I thought of is that if you use an override, you may want to delete the old BTSync override and create the Resilio override from scratch just in case there's an old unchanged variable hanging around there.

Another likely scenario: if you've already got the latest version in your Munki repo (perhaps named BTSync instead of Resilio), you may want to delete that package, rebuild catalogs, and run the Resilio recipe again. The MunkiImporter processor won't import into Munki if the package already exists in your repo.

AutoPkg does have a verbose mode (`-v` used above) which can be helpful in troubleshooting these types of things. AutoPkgr's equivalent is documented [here](https://github.com/lindegroup/autopkgr#step-3-check-the-logs).

**Comment by fseesink** [comment]
You, sir, are a scholar and a prince. :-)  And yes, this was my own stupidity.

First, thank you for the -v option.  I should've thought of that one.  And thanks to that, I noticed while running autopkg that munkiimport was seeing Resilio Sync as already being there.  Long story short, I had removed the cached autokpg folder for the app as well as the munki repo files in pkgs and pkgsinfo and then kept trying the recipe, but all I would see was the app being downloaded but never being added to the repo.  Turns out that I'd neglected to rebuild the catalogs.  I've been so used to using AutoPkgr that it didn't occur to me that I had to rebuild the catalog with

makecatalogs /path/to/repo

Once I did that, things worked as expected.

So again, thanks.  I setup Munki some time ago and haven't had time to add many apps beyond the ~20 I'm currently tracking, and this was the first time I'd dealt with a change in an app's name.  But life's good once again.  Very much appreciate the help.
